addons/account/models/account_move.py

```diff
@@ -286,6 +286,7 @@ def _onchange_journal(self):
 
     @api.onchange('partner_id')
     def _onchange_partner_id(self):
+        self = self.with_company(self.company_id)
```

Comment:
I think more explicit to use `with_company` only when accessing the properties but if you don't agree, it's fine.

Just an idea: Shouldn't be something we could improve ORM-side? Like using `with_company` directly in the `onchange` method?

Reply:
Sthg like if we have a company_id field, do everything in this company ? It would make multicompany features way more stable, but rco doesn't want to (it's a bit too much magic :P, and this would need to be checked in multiple places).

I tend to place the with_company at the beginning of methods, to make sure everything is done in the right company, not only the properties...  We never know when self.env.company will be used to make a domain, define a default or whatever, and it doesn't cost anything with the new cache.

Reply:
So you need to make such similar fix everywhere... everytime... It isn't magic: the multicompany layer is already managed by the ORM (security, access rights, etc)... *sigh*

Reply:
I will maybe couple it with the check_company mechanism, s.t. we force consistency for models requiring consistency.